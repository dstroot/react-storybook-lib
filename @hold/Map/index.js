import React from 'react';

// Import Google Map component
import GoogleMapComponentWithMarker from './GoogleMap';

// Wrapper with Google Map component
const Map = ({ styles, APIKey, scheme }) => (
  <div style={styles}>
    <GoogleMapComponentWithMarker
      googleMapURL={`https://maps.googleapis.com/maps/api/js?v=3.exp&libraries=geometry,drawing,places&key=${APIKey}`}
      loadingElement={<div style={{ height: `100%` }} />}
      containerElement={<div style={{ height: `100%` }} />}
      mapElement={<div style={{ height: `100%` }} />}
    />
  </div>
);

export default Map;

// pass in location, styles, API key and scheme {light or dark}
