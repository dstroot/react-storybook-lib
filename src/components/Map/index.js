import React from "react";
import PropTypes from "prop-types";

// Import Google Map component
import GoogleMapComponentWithMarker from "./GoogleMap";

// Wrapper with Google Map component
const Map = ({ styles, APIKey, scheme }) => (
  <div style={styles}>
    <GoogleMapComponentWithMarker
      scheme={scheme}
      googleMapURL={`https://maps.googleapis.com/maps/api/js?v=3.exp&libraries=geometry,drawing,places&key=${APIKey}`}
      loadingElement={<div style={{ height: `100%` }} />}
      containerElement={<div style={{ height: `100%` }} />}
      mapElement={<div style={{ height: `100%` }} />}
    />
  </div>
);

// note the comments are used for storybook.
Map.propTypes = {
  /** Styles object */
  styles: PropTypes.object.isRequired,
  /** API key for Google maps */
  APIKey: PropTypes.string.isRequired,
  /** Color scheme (light/dark) */
  scheme: PropTypes.string.isRequired
};

export default Map;

// pass in location, styles, API key and scheme {light or dark}
