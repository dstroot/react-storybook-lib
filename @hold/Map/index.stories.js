import React from 'react';
import { storiesOf } from '@storybook/react';

// components
import Map from '../Map';

// Some default styles
const styles = {
  width: '100%',
  height: '500px',
};

const API_KEY = process.env.REACT_APP_GOOGLE_MAPS_API_KEY;

// Map section
storiesOf('Map', module).add('show Map', () => (
  <Map styles={styles} APIKey={API_KEY} scheme="dark" />
));
