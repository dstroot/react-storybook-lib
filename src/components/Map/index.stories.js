import React from 'react';
import { storiesOf } from '@storybook/react';

// components
import Map from '../Map';

// Some default styles
const styles = {
  width: '100%',
  height: '500px',
};

const API_KEY = process.env.REACT_APP_GOOGLE_MAPS_API_KEY;

// Map section
storiesOf('Pac Life|Map', module)
  .addParameters({
    info: {
      inline: true,
      header: false,
    },
  })
  .add('show Map', () => (
    <Map styles={styles} APIKey={API_KEY} scheme="dark" />
  ));
