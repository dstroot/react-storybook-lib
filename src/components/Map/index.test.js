import React from 'react';
import renderer from 'react-test-renderer';
import Map from '../Map';

// Some default styles
const styles = {
  width: '100%',
  height: '500px',
};

const API_KEY = process.env.REACT_APP_GOOGLE_MAPS_API_KEY;

describe('Map', () => {
  it('it should render', () => {
    const component = renderer.create(
      <Map styles={styles} APIKey={API_KEY} scheme="dark" />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
