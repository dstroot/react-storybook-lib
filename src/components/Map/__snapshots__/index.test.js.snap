// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map it should render 1`] = `
<div
  style={
    Object {
      "height": "500px",
      "width": "100%",
    }
  }
>
  <div
    style={
      Object {
        "height": "100%",
      }
    }
  />
</div>
`;
